### PR TITLE
README.md: change web client url to tldr.inbrowser.app

### PR DESCRIPTION
**Pull request by rwv**
Since [ostera/tldr.jsx](https://github.com/ostera/tldr.jsx) is [archived](https://github.com/ostera/tldr.jsx/issues/173#issuecomment-864447513) and lack some modern features. I build this PWA for tldr pages.

Link: <https://tldr.inbrowser.app>

It includes the following features compared to ostera/tldr.jsx

* Autocomplete and search commands.
* Browse by platforms and languages.
* Dark mode support.
* Everything is processed in your browser.
* Works without network connection using PWA.
* All files are static. No backend servers needed.
* Zero API latency, no waiting for response.
* Automatically updates when new tldr-pages are available.

in a nutshell, this site fetch whole repo zip (~3-4MB), decompress and parse and render to markdown all in browser.

- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

### CODE REVIEW

**Review by EmilySeville7cfg — APPROVED** [review]
No fixes found, approved.
Thanks for a contribution! ❤️

File: README.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ If you want an offline version without installing any software,
 check out the [PDF version](https://tldr.sh/assets/tldr-book.pdf).
 
 For browsing without installing a client to your computer,
-see the web client at <https://tldr.ostera.io>.
+see the web client at <https://tldr.inbrowser.app> (offline capable using PWA).
```
Comment (kbdharun) [inline README.md:76]:
```suggestion
see the web client at <https://tldr.inbrowser.app> (with offline support using PWA).
```

**Review by kbdharun — APPROVED** [review]
I am fine with this. Thanks for your contribution. Let's see what others think too.

**Comment by kbdharun** [comment]
Nice, I checked this client just now. Very cool implementation.

**Comment by waldyrious** [comment]
Thank you for the initiative, @rwv! Seems like you did some nice work there!

This is a pretty big (as in significant) change. We should probably consider actually moving tldr-pages.github.io (and tldr.sh) to this implementation in tandem with this PR, leaving @ostera's client live at https://tldr.ostera.io.

I am generally favorable to the idea, but I would like to take a closer look and possibly leave some suggestions. I'd ask maintainers to refrain from merging for a few days (maybe a even a week or two, given the scope of the change) so that others can review and comment. In fact, maybe I'll go ahead and mark this as a draft so that it doesn't accidentally get merged. I hope you don't mind, @rwv!

**Comment by kbdharun** [comment]
> Thank you for the initiative, @rwv! Seems like you did some nice work there!
> 
> This is a pretty big (as in significant) change. We should probably consider actually moving tldr-pages.github.io (and tldr.sh) to this implementation in tandem with this PR, leaving @ostera's client live at https://tldr.ostera.io.
> 
> I am generally favorable to the idea, but I would like to take a closer look and possibly leave some suggestions. I'd ask maintainers to refrain from merging for a few days (maybe a even a week or two, given the scope of the change) so that others can review and comment. In fact, maybe I'll go ahead and mark this as a draft so that it doesn't accidentally get merged. I hope you don't mind, @rwv!

Agreed, that's exactly why I added decision label and requested the first set of reviews.

**Review by mfrw — COMMENTED** [review]
😍  This is amazing :)
Thank you @rwv for your contribution & work.

**Comment by rwv** [comment]
Added some features:

* Printer-friendly: `@media print` hides unwanted elements when printing.
[tar _ tldr InBrowser.App.pdf](https://github.com/tldr-pages/tldr/files/10385095/tar._.tldr.InBrowser.App.pdf)
* `opensearch.xml` enables users to run `tldr tar` in chrome's address bar <https://tldr.inbrowser.app/search/activate-site-search>
* Syntax highlight for placeholder and mnemonics
* Fetch from repo zip and github simultaneously for faster cold start

**Comment by kbdharun** [comment]
Amazing progress, I only had issues with the search bar inside pwa where when I type a command (page) nothing returns and the field becomes empty.

**Review by sbrl — APPROVED** [review]
Woah, that's a pretty slick web app you've got there - fabulous!

As @ waldyrious suggests, we should also update https://github.com/tldr-pages/tldr-pages.github.io/ as well, but that would involve a separate PR, if you wouldn't mind opening it?

This approval is subject to tldr.inbrowser.app being available long-term. If there's a risk it could go down in the future, then we should take steps to avoid this now.

**Comment by rwv** [comment]
> This approval is subject to tldr.inbrowser.app being available long-term. If there's a risk it could go down in the future, then we should take steps to avoid this now.

Since this is a pure front end project and all files are static and use cloudflare pages to serve, there is nearly zero chance it can go down.

**Comment by rwv** [comment]
> As @ waldyrious suggests, we should also update https://github.com/tldr-pages/tldr-pages.github.io/ as well, but that would involve a separate PR, if you wouldn't mind opening it?

PR opened at https://github.com/tldr-pages/tldr-pages.github.io/pull/65

**Comment by ostera** [comment]
Great stuff @rwv 👏🏽 -- I'm happy to see my client superseded by a more modern one.

Didn't check the code but the idea at least has my full support 🙌🏽

**Comment by waldyrious** [comment]
Sorry all for the hold up! :sweat_smile:

I've left a comment in https://github.com/tldr-pages/tldr-pages.github.io/pull/65. But in the meantime, I think we there's no harm in merging this one. Once again, and as others said, the client is really cool and slick. Nice work! :clap:

My main concern is related to the long-term stability of this project, i.e. its dependence on the domain name and the Cloudflare setup. I'm glad to hear that this is a static website with no server-side components. Does that mean one could set it up to be served via GitHub Pages? If that could be done, it would give a bit more tranquility as both the hosting and the domain name would be pretty much guaranteed to stay available, and it would also allow others to experiment with it in a more convenient/beginner-friendly manner (e.g. fork the repo, make a small change and check the result in their own GitHub Pages-published site, all without leaving the browser). WDYT, @rwv?

In any case, I'll mark this PR as ready to merge, as these proposed changes do not affect the fact that it is already an improvement over the current client.

**Review by waldyrious — APPROVED** [review]
LGTM, nice work! I've left some suggestions in the main thread, but they should not block this PR from being merged :)

**Comment by rwv** [comment]
> My main concern is related to the long-term stability of this project, i.e. its dependence on the domain name and the Cloudflare setup. I'm glad to hear that this is a static website with no server-side components. Does that mean one could set it up to be served via GitHub Pages? If that could be done, it would give a bit more tranquility as both the hosting and the domain name would be pretty much guaranteed to stay available, and it would also allow others to experiment with it in a more convenient/beginner-friendly manner (e.g. fork the repo, make a small change and check the result in their own GitHub Pages-published site, all without leaving the browser). WDYT, @rwv?

I still prefer Cloudflare Pages because it offers some awesome features compared to GitHub Pages such as HTTP header modifications, global low latency CDN, br compression for zip, etc. which makes app cold start fast. 


For your concern about stability, Cloudflare Pages is totally free and it offer `pages.dev` domain similar to `github.io` domain. 

If anyone wants to serve site using GitHub Pages, they can write a GitHub Action to publish and there are templates provided by GitHub to do this.

**Comment by waldyrious** [comment]
Sure, I didn't mean to suggest you should change how the project is hosted — your preferences about how to do it are legitimate. I just don't think we (the tldr-pages project) should leave it to potential contributors to figure out how to host a version of this without replicating your setup. I have added an item to my personal tldr-pages todo list to look into that, as IMO it will be helpful both for potential contributors, and as a quick back-up alternative just in case e.g. the domain expires and you happen to be unavailable, or something else goes wrong.

**Comment by rwv** [comment]
https://github.com/InBrowserApp/tldr.inbrowser.app/actions/runs/4089696799

This a GitHub Action which builds the project and uploads artifact to GitHub. You can find the whole static site in `dist` artifact.

**Comment by waldyrious** [comment]
Awesome, thanks! :heart:

I tried tweaking the workflow file ([15a3d33](https://github.com/waldyrious/tldr.inbrowser.app/commit/15a3d33f77ea85a218600b0e77a11253474a75d7), [9043b96](https://github.com/waldyrious/tldr.inbrowser.app/commit/9043b96409739d315729febd02774e0fe25cfba6)) to publish to GitHub Pages, but the published site at https://waldyrious.github.io/tldr.inbrowser.app/ doesn't seem to be working. I suppose I need to do a bit more digging (which is precisely the type of roadblock I mentioned above we ought to ensure contributors don't have to deal with :)).

Btw, if anyone has suggestions for fixing the problem, please let me know! (But again, that is not a blocker for this PR.)

**Comment by rwv** [comment]
You need to put the whole site under path `/` instead of `/tldr.inbrowser.app/`. Otherwise, there will be issues for script importing and vue-router.

**Comment by waldyrious** [comment]
Hmm, I see. Anyway, troubleshooting that would be out of scope for this PR, so let's have that conversation elsewhere. Maybe I'll create an issue in your repo proposing a way for contributors to try out changes to the app in their forks via GitHub Pages. If I manage to get it working well, I would then submit a PR to your repo making changes to the Actions workflow so it will be reusable by others to that end.

**Comment by kbdharun** [comment]
Can this be merged?

**Comment by rwv** [comment]
@CleanMachine1 @navarroaxel @SethFalco Can you review this PR? Thank you!

**Review by SethFalco — APPROVED** [review]

**Review by navarroaxel — APPROVED** [review]
